### DISCUSSION TITLE
Need help with reading a csv

### DISCUSSION THREAD

**Discussion by adriantaw** [grafana/grafana-csv-datasource#52] [Q&A]
Hi, need your help with my csv file.
It works fine with the example that you provided, but it seems like it could not read my csv file.
I got this error "record on line 2: wrong number of fields", in the query.
I check my csv and it seems like everything is ok.
Could you maybe help me to check what is wrong?
[gls3.zip](https://github.com/marcusolsson/grafana-csv-datasource/files/6235981/gls3.zip)

**Comment by marcusolsson** [comment]
The plugin uses the header to determine how many columns to expect (in your case 20 it seems). It seems like there are a few errors in your data set that can't be parsed:

```
CraneBOX action:
- Error code sent to SPOC/AM (CSR)
- Indicate lost SRM communication.",,Crane30,CraneAccalon.CSR.123,54:00.0,561.8959998,54:00.0,CraneAccalon.CSR.123,,,FALSE,,54:00.0,11334919
```
